src/Sulu/Bundle/AdminBundle/Admin/View/ResourceTabViewBuilder.php

```diff
@@ -56,7 +56,7 @@ public function addRouterAttributesToBackView(
 
     public function addRouterAttributesToBlacklist(
         array $routerAttributesToBlacklist
-    ): ResourceTabViewBuilderInterface {
+    ): TabViewBuilderInterface {
```

Comment:
That's wrong, isn't it? This function returns a `ResourceTabViewBuilderInterface`. This way it looks like only methods of the `TabViewBuilderInterface` can be called after calling this function, which is not true.

Reply:
Because of the return type covariance, as you mentioned below.

Reply:
Right, and as mentioned below that is the reason we do not inherit from other interfaces and itnroduced traits in this code :slightly_smiling_face: